o4/o4.py

```diff
@@ -1007,39 +1063,40 @@ def o4_fail():
         if n < 100:
             files.append(f"  {f[F_PATH]}#{f[F_REVISION]}")
 
-    if not files and not passthroughs:
+    if not files and not infos and not warnings and not errors:
         sys.exit(0)
 
-    err_print()
-    hdr = ' o4 ERROR ' if files else ' o4 WARNING '
-    l = (78 - len(hdr)) // 2
-    hdr = '*' * l + hdr + '*' * l
-    ftr = '*' * len(hdr)
-    err_print(f'{CLR}{hdr}')
+    if infos:
+        err_print('*** INFO:\n\t', end='')
+        err_print('\n\t'.join(sorted(infos)))
+
+    if warnings:
+        err_print('*** WARNING:\n\t', end='')
+        err_print('\n\t'.join(sorted(warnings)))
+        if not files and not errors:
+            with open(INCOMPLETE_INDICATOR, 'w') as f:
+                pass
 
     if files:
-        err_print('These files did not sync and are ERRORs')
+        err_print('These files did not sync')
         err_print('\n'.join(sorted(files)))
         if len(files) != n:
             err_print(f'  ...and {n-len(files)} others!')
-        err_print(f'\n{ftr}')
 
-    if passthroughs:
-        err_print('These files did not sync and are WARNINGs')
-        err_print('\n'.join(sorted(passthroughs)))
-        err_print(ftr)
-        if not files:
-            with open(INCOMPLETE_INDICATOR, 'w') as f:
-                pass
-            err_print(
-                f'{CLR} o4 IS EXITING WITH SUCCESS EVEN THOUGH THOSE FILES ARE NOT UP-TO-DATE')
-            err_print(ftr)
-            sys.exit(0)
+    if errors:
+        err_print('*** ERROR:\n\t', end='')
+        err_print('\n\t'.join(sorted(errors)))
+
+    if warnings and not files and not errors:
+        with open(INCOMPLETE_INDICATOR, 'w') as f:
+            pass
+        err_print(f'{CLR} o4 IS EXITING WITH SUCCESS EVEN THOUGH SOME FILES MAY NOT BE UP-TO-DATE')
```

Comment:
This seems like a bad idea, why are why not failing?

Reply:
I think it came from customer screaming, and it's only if there were only warnings and not errors, if that distinction is actually significant.